rustls/src/session.rs

```diff
@@ -231,9 +232,16 @@ pub struct SessionRandoms {
     pub server: [u8; 32],
 }
 
-static TLS12_DOWNGRADE_SENTINEL: &[u8] = &[0x44, 0x4f, 0x57, 0x4e, 0x47, 0x52, 0x44, 0x01];
+pub static TLS12_DOWNGRADE_SENTINEL: &[u8] = &[0x44, 0x4f, 0x57, 0x4e, 0x47, 0x52, 0x44, 0x01];
 
 impl SessionRandoms {
+    pub fn from_client_and_server(we_are_client: bool, client_random: &Random, server_random: &Random) -> Self {
```

Comment:
It looks like the only caller always passes `true` for `we_are_client`. Maybe restrict the API to make this more obvious?

Should the `SessionRandoms::client` and `server` fields just be `Random` rather than untyped arrays? It feelsincongruent to me that you pass in a `&Random` here only to deref it back into an array whereas elsewhere you have `SessionSecrets::for_resume()` start to require the caller to clone.

Reply:
> It looks like the only caller always passes `true` for `we_are_client`. Maybe restrict the API to make this more obvious?

I don't think that will continue to be the case if/when somebody does the server side of this.
 
> Should the `SessionRandoms::client` and `server` fields just be `Random` rather than untyped arrays? It feels incongruent to me that you pass in a `&Random` here only to deref it back into an array whereas elsewhere you have `SessionSecrets::for_resume()` start to require the caller to clone.

I agree, but that was an issue before as well.

If you'd like to make some PRs that refactor the random stuff, I am happy to base this PR on top of them, no matter how disruptive they are to this PR.